### PR TITLE
GCC fix

### PR DESCRIPTION
**Pull request by sarahmarshy**
Added existing o  files to be linked. Added flags for library paths vs library names. Included standard c flags. Gave linker an entry point of main().

### CODE REVIEW

File: project_generator/templates/makefile_gcc.tmpl [hunk 1]
```diff
@@ -44,7 +44,10 @@ LD_SCRIPT = {{linker_file}}
 
 CC_SYMBOLS = {% for symbol in macros %} -D{{symbol}} {% endfor %}
 
-LIBS = {% for library in source_files_lib %} -l{{library}} {% endfor %}
+LIBS = {% for library in libraries %} {{library}} {% endfor %}
+LIB_PATHS = {% for path in lib_paths %} {{path}} {% endfor %}
+LD_SYS_LIBS = -lstdc++ -lsupc++ -lm -lc -lgcc -lnosys
```
Comment (0xc0170) [inline project_generator/templates/makefile_gcc.tmpl]: This is application specific. We should not require any of those in this template.

**Comment by 0xc0170** [comment]
Why there are conflicts?

I would appreciate more descriptive commits. There are many fixes here but not explained at all. Only gcc fix in the commit message.

File: project_generator/templates/makefile_gcc.tmpl (hunk 1 above)
Comment (sarahmarshy) [inline project_generator/templates/makefile_gcc.tmpl]: Should user add them then?

Comment (sarahmarshy) [inline project_generator/templates/makefile_gcc.tmpl]: These are standard flags I thought

Comment (0xc0170) [inline project_generator/templates/makefile_gcc.tmpl]: Yes, a user should. As if someone does not use c++, then first 2 does not make sense for example.

**Comment by 0xc0170** [comment]
Test results - red :(

File: project_generator/templates/makefile_gcc.tmpl (hunk 1 above)
Comment (0xc0170) [inline project_generator/templates/makefile_gcc.tmpl]: Shall we remove it? Then merge

File: project_generator/tools/gccarm.py [hunk 2]
```diff
@@ -13,14 +13,16 @@
...
-from os.path import join, normpath,dirname
 
-from .tool import Tool, Exporter
+from os.path import join, normpath,dirname
+import os
+from .tool import Tool,Exporter
 from ..targets import Targets
+import logging
+import ntpath
+import subprocess
```
Comment (0xc0170) [inline project_generator/tools/gccarm.py]: Please group them as its described: https://www.python.org/dev/peps/pep-0008/#imports

**Pull request merged**